R/spec.R

```diff
@@ -1,5 +1,15 @@
 
 spec <- function(spec, env = caller_env(), signaller = "signal_lifecycle") {
+  if (inherits(spec, "AsIs")) {
+    return(list(
+      fn = spec,
+      arg = NULL,
+      pkg = spec_package(env, signaller = signaller),
```

Comment:
I think this isn't quite right. When you supply a `with` argument to `deprecate_warn()` and friends, `spec()` is called with `env = NULL`, and if you look at the other usage of `spec_package()` below, it isn't supposed to be called with a `NULL` env. `spec_package(NULL)` actually returns the base env because of `topenv(NULL)`, so `"base"` is what gets stored here.

That `pkg = "base"` result never gets used, so it doesn't break anything, but I'd be tempted to just set `pkg = NULL` here to declare that it isn't a field that gets used

Reply:
Update: it looks like `pkg` is being used, see the snapshot tests comment

Reply:
I moved the logic into `spec_package()` which makes it better parallel the other spec helpers.